boards/components/src/date_time.rs

```diff
@@ -67,7 +67,6 @@ impl<D: 'static + date_time::DateTime<'static> + kernel::deferred_call::Deferred
         let grant_date_time = self.board_kernel.create_grant(self.driver_num, &grant_dt);
 
         let date_time = s.write(DateTimeCapsule::new(self.rtc, grant_date_time));
-        kernel::deferred_call::DeferredCallClient::register(self.rtc);
```

Comment:
Shouldn't boards that _do_ use the DateTime driver still have their deferred call be registered through that component? For instance, the Nucleo F429ZI also uses this component and may rely on this line to be present, right?

Reply:
I think that each driver that creates a `DeferredCall` instance needs to register the deferred call. In the case here, the `DateTimeCapsule` does not create a new `DeferredCall` instance and has no idea if the underlying `rtc` actually creates one.

Reply:
The stm32 chip does register its deferred call here https://github.com/tock/tock/blob/9a42b4b356b15e4bf746116aeb9ab2ca82478888/chips/stm32f429zi/src/interrupt_service.rs#L36.

Reply:
Gotcha, this makes sense to me! Just wanted to check that this PR doesn't inadvertently break any other boards.

It seems then that the STM board instantiation worked, but double-registered the deferred call? Looking at the kernel code, that looks like it just works right now. I'm wondering whether it'd make sense to panic in this case (perhaps configurable behind a flag).

Reply:
I think we need to check this in a better way, the bug was very subtle, looks a while to figure it out.